### PR TITLE
add Windows port of perf.c

### PR DESCRIPTION
**Pull request by vtorri**

### CODE REVIEW

File: source/lexbor/ports/windows_nt/lexbor/perf.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    LARGE_INTEGER freq;
+}
+lexbor_perf_t;
+
+
+void *
+lexbor_perf_create(void)
+{
+    lexbor_perf_t *perf = lexbor_calloc(1, sizeof(lexbor_perf_t));
+    if (perf != NULL) {
```
Comment (lexborisov) [inline source/lexbor/ports/windows_nt/lexbor/perf.c]: I think that it should be better to read:
```C
if (perf == NULL) {
    return NULL;
}

/*
 * According to MSDN, QueryPerformanceFrequency() never fails
 * on Windows XP or later
 */
QueryPerformanceFrequency(&perf->freq);

return perf;
```

File: source/lexbor/ports/windows_nt/lexbor/perf.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+lxb_status_t
+lexbor_perf_begin(void *perf)
+{
+    /*
+     * According to MSDN, QueryPerformanceCounter() never fails
+     * on Windows XP or later
+     */
+    QueryPerformanceCounter(&(((lexbor_perf_t *) (perf))->start));
```
Comment (lexborisov) [inline source/lexbor/ports/windows_nt/lexbor/perf.c]: It is better to do so:
```C
lexbor_perf_t *perf_obj = perf;
/*...*/
QueryPerformanceCounter(&perf_obj->start);
```

File: source/lexbor/ports/windows_nt/lexbor/perf.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+lxb_status_t
+lexbor_perf_end(void *perf)
+{
+    /*
+     * According to MSDN, QueryPerformanceCounter() never fails
+     * on Windows XP or later
+     */
+    QueryPerformanceCounter(&(((lexbor_perf_t *) (perf))->start));
```
Comment (lexborisov) [inline source/lexbor/ports/windows_nt/lexbor/perf.c]: Same thing, look above

**Comment by vtorri** [comment]
i do not agree on this : the purpose of the perf counter is to be fast.

for the 1st comment, i use a conditional branch which will almost always be true (if there is no memory for 48 bytes, we are in a very bad shape)
for the 2 other comments, i don't understand the interest of allocating on a stack a variable and using it. I don't  know if the compiler can optimize it and remove the stack variable

on the other hand, i see a bug : the 'start' field is set in lexbor_perf_end() while it should be 'end'

**Comment by lexborisov** [comment]
@vtorri 
My remarks are related solely to the readability of the code.

In the first comment it's better to check that the memory is not allocated and you need to go out. In this entire project, these [checks are used](https://github.com/lexborisov/lexbor/blob/master/source/lexbor/html/tree/insertion_mode/in_body.c#L657).
For the rest of the comments. Not critical. You can leave it as it is, but for better readability ...

**Comment by lexborisov** [comment]
Awesome!
Thanks!

**Pull request merged**